subsys/ipc/rpmsg_multi_instance/Kconfig

```
@@ -0,0 +1,59 @@
+# Copyright (c) 2020-2021 Nordic Semiconductor (ASA)
+# SPDX-License-Identifier: Apache-2.0
+
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_IPC_SHM := zephyr,ipc_shm
+DT_CHOSEN_Z_IPC := zephyr,ipc
+DT_CHOSEN_Z_IPC_TX := zephyr,ipc_tx
+DT_CHOSEN_Z_IPC_RX := zephyr,ipc_rx
+
+menuconfig RPMSG_MULTI_INSTANCE
+	bool "RPMsg multiple instance"
+	select IPM
+	select OPENAMP
+	help
+	  Enables support for RPMsg multiple instance.
+
+if RPMSG_MULTI_INSTANCE
+
+choice RPMSG_ROLE
+	prompt "RPMSG device role"
+	default RPMSG_MULTI_INSTANCE_REMOTE
+
+config RPMSG_MULTI_INSTANCE_REMOTE
+	bool "Remote"
+
+config RPMSG_MULTI_INSTANCE_MASTER
+	bool "Master"
+endchoice
+
+config RPMSG_MULTI_INSTANCES_NO
+	int "Number of RPMSG instances."
+	default 1
+	help
+	  How many instances are to be used.
+
+config RPMSG_MULTI_INSTANCE_SHM_BASE_ADDRESS
+	hex
+	default "$(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_IPC_SHM))"
+	help
+	  This option specifies base address of the memory region to
+	  be used for the OpenAMP IPC shared memory.
+
+config RPMSG_MULTI_INSTANCE_SHM_SIZE
+	hex
+	default "$(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_IPC_SHM))"
+	help
+	  This option specifies size of the memory region to be used
+	  for the OpenAMP IPC shared memory.
+config RPMSG_MULTI_INSTANCE_INIT_PRIORITY
```

Missing newline between entries
```suggestion

config RPMSG_MULTI_INSTANCE_INIT_PRIORITY
```